src/openrct2/ride/gentle/Maze.cpp

```diff
@@ -181,3 +201,132 @@ TRACK_PAINT_FUNCTION get_track_paint_function_maze(int32_t trackType, int32_t di
 
     return maze_paint_setup;
 }
+
+#pragma region Maze Pathfinding
+
+void swap(MazePathfindingEntry& a, MazePathfindingEntry& b)
+{
+    std::swap(a.Raw, b.Raw);
+}
+
+void MazePathfindingEntry::reset()
+{
+    Raw = 0;
+}
+
+void MazePathfindingEntry::free()
+{
+    // Set to the uninitialized value used by the non-maze peep pathfinding
+    Raw = 0xffffffff;
+}
+
+void MazePathfindingEntry::set(const CoordsXY& coords, uint8_t subTileIndex, Direction dirOrigin)
+{
+    assert(coords.x > 0 && coords.y > 0);
+
+    reset();
+    SubTileIdx = subTileIndex;
+    Origin = dirOrigin;
+    XTile = coords.x / COORDS_XY_STEP;
+    YTile = coords.y / COORDS_XY_STEP;
+}
+
+std::tuple<CoordsXY, uint8_t> MazePathfindingEntry::getCoords() const
```

Comment:
Use std::pair for pairs

Reply:
What is the benefit of using pair of tuple?

Reply:
The compiler can optimize std::pair knowing it has two elements potentially into a single register, this doesn't seem to work with tuples.

Reply:
The optimisation problem is a shame. Otherwise `std::pair` would be obsolete since `std::tuple` does the same thing but also allows more things. `std::get<N>` is the way I get items from both pair and tuple.

Reply:
I find tuples to be generally a bad concept, just use a struct if you have more than 2 things. Having std::get<0>(var) is not exactly great code to read and one can only guess what the field could be.

Reply:
You shouldn't need to use get<...> Very often when you could use structured bindings.
```
auto [var1, var2] = functionReturningTupleOrStructOrPair();```

Reply:
This is true but that still means the caller needs to be aware of whats returned, all solved by structs.

Reply:
Fixed by ebb203f.